types/components/buttons.d.ts

```diff
@@ -105,7 +124,10 @@ export interface PayPalButtonsComponentOptions {
     /**
      * Called when the buyer changes their shipping address on PayPal.
      */
-    onShippingChange?: () => void;
+    onShippingChange?: (
+        data: OnShippingChangeData,
+        actions: OnShippingChangeActions
+    ) => Promise<string>;
```

Comment:
Let's have this return `void` to match the Flow types with the source code: https://github.com/paypal/paypal-smart-payment-buttons/blob/2c3407d19789d7c01ac320b8ba4249e8fddd9a5b/src/props/onShippingChange.js#L43

```suggestion
    ) => Promise<void>;
```